clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -114,8 +143,28 @@ public FetchResponse(Struct struct) {
                 int partition = partitionResponseHeader.getInt(PARTITION_KEY_NAME);
                 Errors error = Errors.forCode(partitionResponseHeader.getShort(ERROR_CODE_KEY_NAME));
                 long highWatermark = partitionResponseHeader.getLong(HIGH_WATERMARK_KEY_NAME);
+                long lastStableOffset = INVALID_LSO;
+                if (partitionResponse.hasField(LAST_STABLE_OFFSET_KEY_NAME))
```

Comment:
Thanks for catching this @lindong28. This field won't be used until the transactional code lands. I submitted a PR with tests:

https://github.com/apache/kafka/pull/2737